apps/s_server.c

```diff
@@ -2939,7 +2939,7 @@ static void print_connection_info(SSL *con)
 
     PEM_write_bio_SSL_SESSION(bio_s_out, SSL_get_session(con));
 
-    peer = SSL_get_peer_certificate(con);
+    peer = SSL_get1_peer_certificate(con);
```

Comment:
```suggestion
    peer = SSL_get0_peer_certificate(con);
```
And then delete `X509_free(peer)` below.